torch/distributed/fsdp/_shard_utils.py

```diff
@@ -124,3 +141,28 @@ def _create_chunk_sharded_tensor(
     return ShardedTensor._init_from_local_shards_and_global_metadata(
         local_shards, sharded_tensor_metadata=sharded_tensor_metadata, process_group=pg
     )
+
+
+def _create_chunk_dtensor(
+    tensor: torch.Tensor,
+    rank: int,
+    device_mesh: DeviceMesh,
+) -> DTensor:
+    """
+    Shard a tensor to chunks along the first dimension. The local rank will gets its
+    corresponding chunk as the local tensor to create a DTensor.
+    """
+    shard_placement = DShard(0)
+    tensor_padded_list, pad_sizes = shard_placement._split_tensor(
+        tensor,
+        device_mesh.size(dim=0),
+        with_padding=True,
+        contiguous=True,
+    )
+    unpadded_list = [
```

Comment:
why bother `unpad` the shards when you can simply use `with_padding=False` in the above `_split_tensor` call?